### PR TITLE
doc(datepicker): Datepicker documentation updated

### PR DESCRIPTION
**Pull request by florin-borceanu**
The datepicker documentation was updated to prevent the edge case when the disabled behavior did not work properly.

## PR Checklist

Please check if your PR fulfills the following requirements:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [x] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #3708 

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

### CODE REVIEW

File: src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.html [hunk 1]
```diff
@@ -1,10 +1,12 @@
 <h4>Basic API</h4>
 <p>
     To use the date picker, add the <code class="clr-code">clrDate</code> directive on an
-    <code class="clr-code">input</code> field.
+    <code class="clr-code">input</code> field and add it between <code class="clr-code">clr-date-container</code> tags.
```
Comment (hippee-lee) [inline src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.html]: Would it sound more natural to describe how to use the clrDate directive and then describe its relationship to the parent container? e.g - 

```To use the date picker, add the <code class="clr-code">clrDate</code> directive to an <code class="clr-code">input</code> field.	Then, place the input inside the <code class="clr-code">clr-date-container</code> container element.```

Comment (hippee-lee) [reply src/website/src/app/documentation/demos/datepicker/demos/datepicker-api.demo.html]: Thank you for updating. 
lgtm - please squash your commits and we can get this merged.
